zerver/tests/test_users.py

```diff
@@ -1950,6 +1951,197 @@ def test_email_sent(self) -> None:
         self.assertEqual(msg.reply_to[0], "noreply@testserver")
         self.assertIn("Dear Hamlet,", msg.body)
 
+    # Helper for setting up various types of messages.
+    def setup_user_with_messages(self, user: UserProfile, admin: UserProfile) -> None:
+        self.send_personal_message(user, admin, content="private message 1")
+        self.send_personal_message(user, admin, content="private message 2")
+        self.send_stream_message(user, "Verona", topic_name="topic1", content="stream message 1")
+        self.send_stream_message(user, "Verona", topic_name="topic1", content="stream message 2")
+        self.send_stream_message(user, "Verona", topic_name="topic2")
+        self.send_stream_message(user, "Verona")
+
+        private_stream = self.make_stream("PrivateStream", invite_only=True)
+        self.subscribe(user, private_stream.name)
+        self.send_stream_message(
+            user, private_stream.name, topic_name="topic1", content="private stream message 1"
+        )
+        self.send_stream_message(
+            user, private_stream.name, topic_name="topic2", content="private stream message 2"
+        )
+
+    def test_api_with_no_delete_action(self) -> None:
```

Comment:
Test functions for message deletion should mention about deactivation to make it more clear like `test_deactivation_without_deleting_messages`, `test_deactivation_with_deleting_public_stream_messages`, etc.